server/repositories/cmsQueries/seriesPageQuery.js

```diff
@@ -0,0 +1,81 @@
+/* eslint-disable class-methods-use-this */
+const { DrupalJsonApiParams: Query } = require('drupal-jsonapi-params');
+const { typeFrom } = require('../../utils/adapters');
+
+class SeriesPageQuery {
+  static #TILE_FIELDS = [
+    'drupal_internal__nid',
+    'title',
+    'field_moj_description',
+    'field_moj_thumbnail_image',
+    'field_moj_series',
+  ];
+
+  constructor(establishmentName, id) {
+    this.establishmentName = establishmentName;
+    this.id = id;
+    this.query = new Query()
+      .addFilter('field_moj_series.meta.drupal_internal__tid', id)
+      .addFields('node--page', SeriesPageQuery.#TILE_FIELDS)
+      .addFields('node--moj_video_item', SeriesPageQuery.#TILE_FIELDS)
+      .addFields('node--moj_radio_item', SeriesPageQuery.#TILE_FIELDS)
+      .addFields('moj_pdf_item', SeriesPageQuery.#TILE_FIELDS)
+      .addFields('file--file', ['image_style_uri'])
+      .addFields('taxonomy_term--series', [
+        'name',
+        'description',
+        'drupal_internal__tid',
+        'field_featured_image',
+      ])
+      .addInclude([
+        'field_moj_thumbnail_image',
+        'field_moj_series.field_featured_image',
+      ])
+      .addSort('series_sort_value', 'ASC')
+      .getQueryString();
+  }
+
+  path() {
+    return `/jsonapi/prison/${this.establishmentName}/node?${this.query}`;
+  }
+
+  #getSeries = item => ({
+    id: item?.drupalInternal_Tid,
+    contentType: 'series',
+    name: item?.name,
+    description: item?.description?.processed,
+    image: {
+      url: item?.fieldFeaturedImage?.imageStyleUri[0]?.tile_large,
+      alt: item?.fieldFeaturedImage?.resourceIdObjMeta?.alt,
+    },
+  });
+
+  #getContent = item => {
+    const id = item?.drupalInternal_Nid;
+    return {
+      id,
+      contentType: typeFrom(item?.type),
+      title: item?.title,
+      summary: item?.fieldMojDescription?.summary,
+      contentUrl: `/content/${id}`,
+      image: {
+        url: item?.fieldMojThumbnailImage?.imageStyleUri[1].tile_small,
+        alt: item?.fieldMojThumbnailImage?.resourceIdObjMeta?.alt,
+      },
+    };
+  };
+
+  transform(deserializedResponse) {
+    return Object.assign(
```

Comment:
Meh.. I was doing a bit of reading around this.. It doesn't look like there is a good reason to use `Object.assign` for places where object spread would suffice. 

As well as spread being more succinct, I think it's generally good to prefer approaches that don't mutate existing data.

The only reason to use Object.assign seems to be if you want to actually call setters or preserve the prototype of the original object - would possibly only reserve it to be used in those cases.

Reply:
The use of spread or assign here is negligible. The first object gets created and torn down and then recreated in this scenario with a spread, theoretically it only gets created once with Object.assign. I'm easy either way though 🤷